Src/Extern/HYPRE/AMReX_HypreSolver.H

```diff
@@ -254,7 +255,13 @@ HypreSolver<MSS>::HypreSolver (Vector<IndexType>   const& a_index_type,
     }
     AMREX_ASSERT(proc_end == proc_begin + m_nrows_proc);
 
-    fill_global_id();
+    // To generate global ids for Hypre, we need to remove duplicates on
+    // nodes shared by multiple Boxes with OverrideSync.  So we need to use
+    // a type that supports atomicAdd.  HYPRE_Int is either int or long
+    // long.  The latter does not have native atomicAdd support in CUDA/HIP.
```

Comment:
Maybe a last sentence, that `unsigned long long` is supported on CUDA/HIP, would describe the new logic fully.